runtime/src/accounts_db.rs

```diff
@@ -3716,9 +3735,14 @@ impl AccountsDB {
         mut hashes: Vec<(Pubkey, Hash, u64)>,
         slot: Slot,
         debug: bool,
+        unstable_sort: bool,
     ) -> ((Hash, u64), (Measure, Measure)) {
         let mut sort_time = Measure::start("sort");
-        Self::sort_hashes_by_pubkey(&mut hashes);
+        if unstable_sort {
+            hashes.par_sort_unstable_by(|a, b| a.0.cmp(&b.0));
+        } else {
+            Self::sort_hashes_by_pubkey(&mut hashes);
```

Comment:
I think all callers have unique pubkeys so stable sort wouldn't make much difference here. Can we just use unstable for all?

Reply:
results are equivalent, but performance is not. When called from the normal code path, I think the pub keys may be mostly or completely sorted since they came from the index. In that case, unstable sort was 3x slower than the stable sort we were using prior to my changes.
However, for the completely unsorted pubkeys that come from searching the stores, the stable sort was ~2x slower than the unstable sort. I read the docs for unstable and stable and it indicated you could get better sorting performance in some conditions with unstable sort, and I did see that.

Reply:
Reworking the pipeline of store searching through sort is one of the perf optimizations that I am aiming to work on next. In the meantime, if I use stable sort or unstable, the performance of something will get much worse.

Reply:
and, yes, all known, current callers should have unique pub keys.

Reply:
I see, we should at least add a comment about that because you would expect unstable sort to be strictly better than stable sort.

And it's consistent with randomized keys?

Reply:
[Here's](https://docs.rs/rayon/1.5.0/rayon/slice/trait.ParallelSliceMut.html#method.par_sort_unstable_by) where I got the idea.
"Unstable is generally faster than stable sorting, except in a few special cases, e.g. when the slice consists of several concatenated sorted sequences."

So, it seems we may want to prefer unstable in this case. However, switching the existing code path to stable resulted in measurably worse performance. I intend to refactor the pipeline of how we calculate a hash based on store data only (for better performance) once this pr goes in. I believe this sort change may disappear at that point. In the meantime, we end up with worse performance in one of the two cases. Here are some numbers that aren't quite what I'd want to compare, but they are perhaps useful:
sort time, unstable, not using index, in ahv threads: 989341us, 14518391 hashes
sort time, stable, using index, in normal validator threads(?): sort=34271us, hash_total=13550803

In my earlier experiments, using unstable sort on data that was indexed resulted in hash times that were much higher that 34ms.

Reply:
Ok, let's add comment. Reading the code it's a bit weird and not obviously done for performance or functionality.